Modules/Core/Common/src/itkPoolMultiThreader.cxx

```diff
@@ -193,15 +200,19 @@ PoolMultiThreader ::ParallelizeArray(SizeValueType             firstIndex,
     ProgressReporter reporter(filter, 0, workUnit);
 
     // execute this thread's share
-    ExceptionHandler exceptionHandler([lambda, firstIndex, chunkSize] { lambda(firstIndex, firstIndex + chunkSize); });
-    reporter.CompletedPixel();
+    ExceptionHandler exceptionHandler;
+    exceptionHandler.TryAndCatch([lambda, firstIndex, chunkSize, &reporter] {
+      lambda(firstIndex, firstIndex + chunkSize);
+      reporter.CompletedPixel();
+    });
 
     // now wait for the other computations to finish
     for (SizeValueType i = 1; i < workUnit; i++)
     {
-      reporter.CompletedPixel();
-
-      exceptionHandler.TryToGetFuture(m_ThreadInfoArray[i]);
```

Comment:
We can probably remove the last use of `TryToGetFuture` and replace it by a call to  more general `TryAndCatch`.

Reply:
done